05-how-to-contribute-to-open-source.md

```diff
@@ -25,139 +29,292 @@ OpenSauced is a powerful tool for finding open source projects to contribute to.
+2. Pull the latest changes with this command:
 
-Congratulations! You just made an open source contribution! In the next chapter, we'll add this to an open source portfolio: a resume for open source contributions through which you can showcase your work and attract potential employers and friends using OpenSauced. For now, let's wrap up with some notes on onboarding.
+   ```bash
+   git pull origin main
+   ```
 
 ## Merge Conflicts
 
-### Understanding Merge Conflicts
+Merge conflicts are something you'll commonly encounter when contributing to an open source project. When two branches have made different changes to the same line(s) in the same file(s), Git cannot automatically determine which change to keep, resulting in a conflict.
+
+When a merge conflict occurs, Git adds conflict markers (`<<<<<<<`, `=======`, and `>>>>>>`) to indicate the conflicting lines from different branches. Everything between the `<<<<<<<` and `=======` is the changes that you worked on (current changes). And everything between the `=======` to `>>>>>>>` is the incoming changes from the remote `main` branch.
+
+You need to pay attention to the conflicts and decide how you want to resolve them. You can keep only your change, incoming change, or both changes.
+
+### Tips to Avoid Further Conflicts
+
+Some open source repositories, such as OpenSauced's [guest book](https://github.com/open-sauced/guestbook) and [pizza verse](https://github.com/open-sauced/pizza-verse) repositories, have high contribution activities in the same files that can cause merge conflicts.
+
+Below are some tips to avoid further conflicts when contributing to open source projects:
+
+#### Following Instructions
+
+Ensure you follow the instructions in the project's README or Contributing Guide, and don't miss any step.
 
-- Merge conflicts happen when two branches have made different changes to the same lines in a file. Git cannot automatically determine which change to keep, and this results in a conflict.
-- When a merge conflict occurs, Git adds special markers (<<<<<<<, =======, and >>>>>>) to the conflicted part of the file, indicating the conflicting sections from different branches.
+#### Pull Request Form
 
-### Resolving Merge Conflicts in VS Code
+Complete the template form and fill in all areas when creating a pull request.
 
-#### Prerequisites
+If a repository doesn't provide you with a pull request template, you need to have these in your pull request form:
 
-Ensure you have [VS Code](https://code.visualstudio.com/) installed on your computer and the latest version of [Git](https://git-scm.com/).
+- A descriptive title
+- A clear description of your changes
```

Comment:
[suggestion] @adiati98, I added some information of how to write a description of a contribution and a real-life example. This is another issue I see amongst beginners these explanations would help them understand how to write this piece of information.  
```suggestion
- A clear description of your changes: This gives contributors and maintainers insight on how you made the changes. Here's [an example of a clear description of a contribution in a PR](https://github.com/open-sauced/intro/pull/10)
```